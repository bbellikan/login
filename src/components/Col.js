import React, { Component } from 'react';
import CellHeader from './CellHeader';
import Editable from './Editable'

class Col extends Component {
    constructor(props) {
        super(props)

        this.getColStyle = this.getColStyle.bind(this);
    }

    getColStyle() {
        return {
            width: this.props.gridCol.width + 'px',
            order: this.props.gridCol.order
        }
    }

    render() {
        return (
            <div className="grid-col"
                style={this.getColStyle()}
            >

                <CellHeader 
                    gridCol={this.props.gridCol}
                    clicked={this.props.clicked}
                    updateClicked={this.props.updateClicked}
                    onRightEdge={this.props.onRightEdge}
                    updateColOrder={this.props.updateColOrder}
                    updateSort = {this.props.updateSort}
                    deleteCol = {this.props.deleteCol}

                />
                {this.props.gridData.map((item,index)=>{
                    return(
                    <Editable
                        inputClassName="input"
                        key={this.props.gridCol.number + '_' + index}
                        colNumber={this.props.gridCol.number}
                        gridCol={this.props.gridCol}
                        rowNumber={index}
                        value = {item[this.props.gridCol.header]}
                        updateCell = {this.props.updateCell}
                    >

                    </Editable>
                    )
                })}

            </div>
        );
    }
}

export default Col;
