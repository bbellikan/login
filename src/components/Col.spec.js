import React from 'react';
import { shallow } from 'enzyme';
import Col from './Col';

const setup = (setupProps = {}) => {
    const defaultProps = {
        gridData : [{
            type: 'POS',
            extdesc: 'ABC VONS Store 1797 SAN DIEGO CAUS',
            description: 'POS Transaction VONS Store 1797 SAN DIEGO CAUS',
            amount: '-4.43',
            differed: false,
            date: '2016-02-23'
        }],
        gridCol:{
            "number": 0,
            "order": 0,
            "header": "type",
            "sort": "n",
            "deleted": false,
            "type": "STRING",
            "inputType": "text",
            "width": 150
        },
        updateClicked:jest.fn(),
        onRightEdge:jest.fn(),
        updateColOrder:jest.fn(),
        updateCell:jest.fn(),
        updateSort : jest.fn(),
        deleteCol : jest.fn(),
        deleteRow:jest.fn(),
    };


    const props = {...defaultProps, ...setupProps};
    const wrapper = shallow(
        <Col {...props}/>
    );

    return {
        props,
        wrapper
    };
};

describe('Col', () => {
    it('renders without crashing', () => {
        const {wrapper} = setup();
        expect(wrapper).toMatchSnapshot();
    });

});
