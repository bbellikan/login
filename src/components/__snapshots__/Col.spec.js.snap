// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Col
    deleteCol={[Function]}
    deleteRow={[Function]}
    gridCol={
        Object {
            "deleted": false,
            "header": "type",
            "inputType": "text",
            "number": 0,
            "order": 0,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          }
    }
    gridData={
        Array [
            Object {
              "amount": "-4.43",
              "date": "2016-02-23",
              "description": "POS Transaction VONS Store 1797 SAN DIEGO CAUS",
              "differed": false,
              "extdesc": "ABC VONS Store 1797 SAN DIEGO CAUS",
              "type": "POS",
            },
          ]
    }
    onRightEdge={[Function]}
    updateCell={[Function]}
    updateClicked={[Function]}
    updateColOrder={[Function]}
    updateSort={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CellHeader
          deleteCol={[Function]}
          gridCol={
                    Object {
                              "deleted": false,
                              "header": "type",
                              "inputType": "text",
                              "number": 0,
                              "order": 0,
                              "sort": "n",
                              "type": "STRING",
                              "width": 150,
                            }
          }
          onRightEdge={[Function]}
          updateClicked={[Function]}
          updateColOrder={[Function]}
          updateSort={[Function]}
/>,
        Array [
          <Editable
            gridCol={
                        Object {
                                    "deleted": false,
                                    "header": "type",
                                    "inputType": "text",
                                    "number": 0,
                                    "order": 0,
                                    "sort": "n",
                                    "type": "STRING",
                                    "width": 150,
                                  }
            }
            rowNumber={0}
            updateCell={[Function]}
            value="POS"
/>,
        ],
      ],
      "className": "grid-col",
      "style": Object {
        "order": 0,
        "width": "150px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "deleteCol": [Function],
          "gridCol": Object {
            "deleted": false,
            "header": "type",
            "inputType": "text",
            "number": 0,
            "order": 0,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "onRightEdge": [Function],
          "updateClicked": [Function],
          "updateColOrder": [Function],
          "updateSort": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0_0",
        "nodeType": "class",
        "props": Object {
          "gridCol": Object {
            "deleted": false,
            "header": "type",
            "inputType": "text",
            "number": 0,
            "order": 0,
            "sort": "n",
            "type": "STRING",
            "width": 150,
          },
          "rowNumber": 0,
          "updateCell": [Function],
          "value": "POS",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CellHeader
            deleteCol={[Function]}
            gridCol={
                        Object {
                                    "deleted": false,
                                    "header": "type",
                                    "inputType": "text",
                                    "number": 0,
                                    "order": 0,
                                    "sort": "n",
                                    "type": "STRING",
                                    "width": 150,
                                  }
            }
            onRightEdge={[Function]}
            updateClicked={[Function]}
            updateColOrder={[Function]}
            updateSort={[Function]}
/>,
          Array [
            <Editable
              gridCol={
                            Object {
                                          "deleted": false,
                                          "header": "type",
                                          "inputType": "text",
                                          "number": 0,
                                          "order": 0,
                                          "sort": "n",
                                          "type": "STRING",
                                          "width": 150,
                                        }
              }
              rowNumber={0}
              updateCell={[Function]}
              value="POS"
/>,
          ],
        ],
        "className": "grid-col",
        "style": Object {
          "order": 0,
          "width": "150px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "deleteCol": [Function],
            "gridCol": Object {
              "deleted": false,
              "header": "type",
              "inputType": "text",
              "number": 0,
              "order": 0,
              "sort": "n",
              "type": "STRING",
              "width": 150,
            },
            "onRightEdge": [Function],
            "updateClicked": [Function],
            "updateColOrder": [Function],
            "updateSort": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0_0",
          "nodeType": "class",
          "props": Object {
            "gridCol": Object {
              "deleted": false,
              "header": "type",
              "inputType": "text",
              "number": 0,
              "order": 0,
              "sort": "n",
              "type": "STRING",
              "width": 150,
            },
            "rowNumber": 0,
            "updateCell": [Function],
            "value": "POS",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
